orderedmap_test.go

```diff
@@ -263,6 +263,19 @@ func TestMove(t *testing.T) {
 	assert.NotEqual(t, err, nil)
 }
 
+func TestNewEdgeCases(t *testing.T) {
+	zero := New[int, string](0)
+	assert.Empty(t, zero.Len())
+
+	assert.PanicsWithValue(t, "too many arguments to New[K,V]()", func() {
+		_ = New[int, string](1, 2, 3)
+	})
+
+	om := New[int, string](-1)
+	om.Set(1337, "quarante-deux")
+	assert.Equal(t, 1, om.Len())
+}
```

Comment:
Would be nice to add a test with a > 0 capacity that also checks that `om.pairs` has the expected capacity.

Reply:
Interesting 👍